File: code/mathematical_algorithms/src/amicable_numbers/amicable_numbers.cpp
// Part of Cosmos by OpenGenus Foundation

//Program to Calculate Amicable Numbers under 10000 and find sum of them

#include <iostream>

using namespace std;

int main()
{
    int n, k;
    int i = 1, s1 = 0, s2 = 0, sum = 0;

    for (k = 1; k <= 10000; k++)
    {
        n = k;
        while (i < n)
        {
            if (n % i == 0)
                s1 = s1 + i;
            i++;
        }
        i = 1;
        if (s1 == n)
            continue;
        while (i < s1)
        {
            if (s1 % i == 0)
                s2 = s2 + i;
            i++;
        }
        if (n == s2)
        {
            cout << n << endl;
            sum += n;
        }

        s1 = 0;
        s2 = 0;
    }
    cout << "\nSum of Amicable numbers under 10000 is : " << sum;
}

File: code/mathematical_algorithms/src/amicable_numbers/amicable_numbers.js
function check(a, b) {
  let sum = 0;
  for (let i = 1; i < a; i++) {
    if (a % i == 0) {
      sum += i;
    }
  }
  if (sum == b) {
    return true;
  }
}

function ami_check(a, b) {
  if (check(a, b)) {
    if (check(b, a)) {
      console.log("Numbers are Amicable");
    } else {
      console.log("Numbers are not Amicable");
    }
  } else {
    console.log("Numbers are not Amicable");
  }
}

ami_check(17296, 18416);
